doc/rst/tools/mason/mason.rst

```diff
@@ -0,0 +1,339 @@
+        
+For example, after ``mason run MyPackage``, the project directory appears as so::
+
+
+    MyPackage/
+      Mason.toml
+      Mason.lock
+      src/
+	MyPackage.chpl
+      target/
+	debug/
+      	release/
+	  benches/
+	  examples/
+	  MyPackage
+	  test/
+   
+
+
+
+
+
+The Design of Mason
+===================
+
+mason: Command line tool for building chapel programs to provide users with 
+       a consistent way of building applications and libraries. Mason uses 
+       a four state pipeline to go from start to finish in a project. The
+       four states are listed below.
+    
+
+Four States:
+
+1) Project Code: ``yourProject/src/yourProject.chpl``
+   	   This is the source code of the project the user creates using mason.
+2) Manifest File: ``Mason.toml``
+           Toml file containing metadata and dependencies
+           Builds dependency directed acyclic graph (DAG) to be
+	   serialized into lock file
+3) Lock File:  ``Mason.lock``
+           Contains necessary build information
+           Serialized directed acyclic graph of the dependencies build options 
+	   from the manifest
+4) Dependency Code:  ``$HOME/.mason/src``
+	   Local dependencies downloaded by mason after the user lists them in 
+	   a project manifest.
+
+
+
+
+
+
+The Manifest File
+=================
+
+The ``Mason.toml`` manifest file is written in TOML(for more information see TOML section below).
+Each time a new project is created in Mason a standard TOML file in included in the top-level
+directory of the project directory. 
+
+For example, ``Mason.toml``:
+
+.. code-block:: text
+
+    [brick]
+    name = "hello_world"
+    version = "0.1.0"
+    authors = ["Bradford Chamberlain <brad@chamberlain.com>"]
+
+    [dependencies]
+    curl = '1.0.0'
+
+
+
+
+
+TOML
+====
+
+TOML is the configuation language chosen by the chapel developers for
+configuring programs written in chapel using mason. A TOML file contains
+the nessescary information to build a chapel program using mason. 
+`TOML Spec <https://github.com/toml-lang/toml>`_.
+
+
+
+
+
+Mason-Registry
+==============
+
+The initial mason registry is a GitHub repository containing a list of versioned manifest files.
+This is not unlike that of the OS X Homebrew package manager registry.
+
+The registry would follow a hierarchy as follows:
+
+
+.. code-block:: text
+
+ registry/
+   Curl/
+      1.0.0.toml
+      2.0.0.toml
+   RecordParser/
+      1.0.0.toml
+      1.1.0.toml
+      1.2.0.toml
+   VisualDebug/
+      2.2.0.toml
+      2.2.1.toml
+
+Each versioned manifest file would be identical to the manifest file in the top-level directory
+of the package repository, with one additional field not required in the repository manifest,
+a URL pointing to the repository and revision in which the version is located.
+
+Continuing the example from before, the 'registry' ``Mason.toml`` would include the additional source field:
+
+.. code-block:: text
+
+     [brick]
+     name = "hello_world"
```

Comment:
Is it intentional that the name ``hello_world`` here doesn't match the ``MyProject`` above?

I gather ``brick`` is a required section, as is ``root`` in the lock file below?